### PR TITLE
Fix unclosed rocks iterators

### PR DESCRIPTION
**Pull request by xandkar**

### CODE REVIEW

File: src/blockchain.erl [hunk 1]
```diff
@@ -892,12 +892,18 @@ find_first_height_after(MinHeight0, #blockchain{db=DB, heights=HeightsCF}) ->
...
+        case rocksdb:iterator_move(Iter, next) of
+            {ok, <<Height:64/integer-unsigned-big>>, Hash} ->
+                {ok, Height, Hash};
+            {error, invalid_iterator} ->
+                {error, not_found};
+            {error, Reason} ->
+                lager:error("Unexpected iterator error: ~p", [Reason]),
+                {error, not_found}
+        end,
+    catch rocksdb:iterator_close(Iter),
```
Comment (jeffgrunewald) [inline src/blockchain.erl:905]: thought we were trying to avoid these types of catches since they swallowed errors

Comment (xandkar) [reply src/blockchain.erl:905]: Yeah. I hate it. I've done a bunch of rocks iteration experiments recently and stumbled on enough seemingly-nondeterministic iterator-access errors to make me nervous.

While I lean on removing these `catch`s here, I'm not sure what will happen if you try to close an iterator after an error other than `invalid_iterator` - hence the recon logging.

I wasn't able to reproduce those not-`invalid_iterator` errors.

Comment (xandkar) [reply src/blockchain.erl]: @jeffgrunewald OK, I replaced all the `catch rocksdb:iterator_close`s with `try rocksdb:iterator_close catch log end`s.

**Review by jeffgrunewald — APPROVED** [review]
Love the macro!

File: src/blockchain_rocks.hrl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+-define(ROCKSDB_ITERATOR_CLOSE(IteratorHandle),
+    %% Wrapping in fun to avoid scope leakage:
+    (fun() ->
+        try
+            rocksdb:iterator_close(IteratorHandle)
+        catch ErrorClass:ErrorReason ->
+            lager:error("rocksdb:iterator_close error: ~p:~p", [ErrorClass, ErrorReason])
```
Comment (evanmcc) [inline src/blockchain_rocks.hrl:7]: change to warning, please

**Comment by evanmcc** [comment]
suspect that the catch logging is going to be noisy, we might want to add a version of the macro that allows us not to log.

in any case, we want to log at `warning`, since error writes to the error log with `fsync`

File: src/blockchain_rocks.hrl (hunk 2 above)
Comment (xandkar) [reply src/blockchain_rocks.hrl]: done.

**Comment by xandkar** [comment]
> suspect that the catch logging is going to be noisy, we might want to add a version of the macro that allows us not to log.

added the an an app env var to disable the logging.

> in any case, we want to log at `warning`

switched it to warning.

> since error writes to the error log with `fsync`

TIL!

**Pull request merged**